libclamav_rust/src/util.rs

```diff
@@ -46,3 +48,23 @@ pub fn file_from_fd_or_handle(fd: i32) -> File {
     #[cfg(not(any(windows, unix)))]
     compile_error!("implemented only for unix and windows targets")
 }
+
+/// Get a string from a pointer
+///
+/// # Safety
+///
+/// The caller is responsible for making sure the lifetime of the pointer
+/// exceeds the lifetime of the output string.
+pub fn optional_cstring(ptr: *const i8) -> Option<&'static str> {
```

Comment:
This is kind of a weird semantic. You're doing the Result->Option conversion for the caller, and denying them the error (if they want it). They could do that themselves with `.ok()`

Maybe deserves a second look.  This could be genericized a bit as `cstr_from_ptr`, and the calls might read a little better.

Reply:
Good catch on the enumerate, but regarding `all()`, no we return `false` if there is an error.